src/Config/RuntimeConfig.cs

```diff
@@ -1,383 +1,126 @@
-            {
-                string errorMessage = $"Deserialization of the configuration file failed.\n" +
-                        $"Message:\n {ex.Message}\n" +
-                        $"Stack Trace:\n {ex.StackTrace}";
-
-                if (logger is null)
-                {
-                    // logger can be null when called from CLI
-                    Console.Error.WriteLine(errorMessage);
-                }
-                else
-                {
-                    logger.LogError(errorMessage);
-                }
-
-                deserializedRuntimeConfig = null;
-                return false;
-            }
-        }
-
-        [JsonIgnore]
-        public RestGlobalSettings RestGlobalSettings { get; private set; } = new();
-
-        [JsonIgnore]
-        public GraphQLGlobalSettings GraphQLGlobalSettings { get; private set; } = new();
-
-        [JsonIgnore]
-        public HostGlobalSettings HostGlobalSettings { get; private set; } = new();
-
-        [JsonIgnore]
-        public Dictionary<string, string> GraphQLSingularTypeToEntityNameMap { get; private set; } = new();
-
-        public bool IsEasyAuthAuthenticationProvider()
-        {
-            // by default, if there is no AuthenticationSection,
-            // EasyAuth StaticWebApps is the authentication scheme.
-            return AuthNConfig != null &&
-                   AuthNConfig.IsEasyAuthAuthenticationProvider();
-        }
-
-        public bool IsAuthenticationSimulatorEnabled()
-        {
-            return AuthNConfig != null &&
-                AuthNConfig!.IsAuthenticationSimulatorEnabled();
+                GraphQL = nameCorrectedEntity.GraphQL
+                with
+                { Plural = nameCorrectedEntity.GraphQL.Singular.Pluralize() }
+            };
         }
 
-        public bool IsJwtConfiguredIdentityProvider()
+        // If this is a Stored Procedure with no provided GraphQL operation, set it to Mutation as the default
+        if (nameCorrectedEntity.GraphQL.Operation is null && nameCorrectedEntity.Source.Type is EntityType.StoredProcedure)
         {
-            return AuthNConfig != null &&
-                !AuthNConfig.IsEasyAuthAuthenticationProvider() &&
-                !AuthNConfig.IsAuthenticationSimulatorEnabled();
+            nameCorrectedEntity = nameCorrectedEntity
+                with
+            { GraphQL = nameCorrectedEntity.GraphQL with { Operation = GraphQLOperation.Mutation } };
         }
 
-        [JsonIgnore]
-        public DatabaseType DatabaseType
+        // If no Rest node was provided in the config, set it with the default state of enabled for all verbs
+        if (nameCorrectedEntity.Rest is null)
         {
-            get
-            {
-                return DataSource.DatabaseType;
-            }
+            nameCorrectedEntity = nameCorrectedEntity
+                with
+            { Rest = new EntityRestOptions(EntityRestOptions.DEFAULT_SUPPORTED_VERBS) };
```

Comment:
Yes, I'll pull it out.

It's only in there because I started with GraphQL defaults, then realised there were some REST ones to add and didn't pay attention to the method name.